examples/Bickley_jet_shallow_water.jl

```diff
@@ -0,0 +1,244 @@
+# ``uh`` and ``vh`` in the ``x`` and ``y`` directions, respectively,
+# and the height ``h``.  Note that ``u`` and ``v`` are the velocities, which 
+# can easily be computed when needed.
+#
+# ## Install dependencies
+#
+# First we make sure that we have all of the packages that are required to
+# run the simulation.
+# ```julia
+# using Pkg
+# pkg"add Oceananigans, NCDatasets, Plots, Printf, Polynomials"
+# ```
+#
+
+using Oceananigans
+using Oceananigans.Models: ShallowWaterModel
+
+# ## Two-dimensional domain 
+#
+# The shallow water model is a two-dimensional model and we must specify
+# the number of vertical grid points to be one.  
+# We pick the length of the domain to fit one unstable mode along the channel
+# exactly.  Note that ``Lz`` is the mean depth of the fluid.  It is determined
+# using linear stability theory that the zonal wavenumber that yields the most unstable
+# mode that fits in this zonal channel has a growth rate of ``\sigma \approx 0.14``.
+
+Lx = 2π
+Ly = 20
+Lz = 1
+Nx = 128
+Ny = Nx
+
+grid = RegularRectilinearGrid(size = (Nx, Ny, 1),
+                            x = (0, Lx), y = (0, Ly), z = (0, Lz),
+                            topology = (Periodic, Bounded, Bounded))
+
+# ## Physical parameters
+#
+# This is a toy problem and we choose the parameters so this jet idealizes
+# a relatively narrow mesoscale jet.   
+# The physical parameters are
+#
+#   * ``f``: Coriolis parameter
+#   * ``g``: Acceleration due to gravity
+#   * ``U``: Maximum jet speed
+#   * ``\Delta\eta``: Maximum free-surface deformation that is dictated by geostrophy
+#   * ``\epsilon`` : Amplitude of the perturbation
+
+f = 1
+g = 9.80665
+U = 1.0
+Δη = f * U / g
+ϵ = 1e-4
+nothing # hide
+
+# ## Building a `ShallowWaterModel`
+#
+# We use `grid`, `coriolis` and `gravitational_acceleration` to build the model.
+# Furthermore, we specify `RungeKutta3` for time-stepping and `WENO5` for advection.
+
+model = ShallowWaterModel(
+    timestepper=:RungeKutta3,
+    advection=WENO5(),
+    grid=grid,
+    gravitational_acceleration=g,
+    coriolis=FPlane(f=f),
+    )
+
+# ## Background state and perturbation
+# 
+# We specify `Ω` to be the background vorticity of the jet in the absence of any perturbations.
+# The initial conditions include a small-ampitude perturbation that decays away from the center
+# of the jet.
+
+  Ω(x, y, z) = 2 * U * sech(y - Ly/2)^2 * tanh(y - Ly/2)
+ uⁱ(x, y, z) =   U * sech(y - Ly/2)^2 + ϵ * exp(- (y - Ly/2)^2 ) * randn()
+ ηⁱ(x, y, z) = -Δη * tanh(y - Ly/2)
+ hⁱ(x, y, z) = model.grid.Lz + ηⁱ(x, y, z)
+uhⁱ(x, y, z) = uⁱ(x, y, z) * hⁱ(x, y, z)
+nothing # hide
+
+# We set the initial conditions for the zonal mass transport `uhⁱ` and height `hⁱ`.
+
+set!(model, uh = uhⁱ , h = hⁱ)
+
+# We compute the total vorticity and the perturbation vorticity.
+
+uh, vh, h = model.solution
+        u = ComputedField(uh / h)
+        v = ComputedField(vh / h)
+  ω_field = ComputedField( ∂x(vh/h) - ∂y(uh/h) )
```

Comment:
Can we use `ComputedField(∂x(v) - ∂y(u))` here?

Reply:
This is that, since `v = vh / h`. `v` is not a prognostic field in `ShallowWaterModel`.

Reply:
So you cannot define `ComputedField`s via an already defined `ComputedField`?

Reply:
Btw, @francispoulin, why we call it `ω_field` and not simply `ω` for symmetry reasons?
Or call also `u_field`, `v_field`?... nah...

Also, again for symmetry reasons and to satisfy my ocd: `= ComputedField(∂x(vh / h) - ∂y(uh / h))`, i.e. no spaces before/after parentheses but spaces surrounding `/`.

Reply:
All good points, as always, and will make the changes.  I tried replacing `vh/h` with v and it doesn't work, not sure why.  So I'll keep things as they were.

Reply:
Sadly it doesn't work for me but I made all the other changes you suggested, which are better